simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/BasicAuthenticator.java

```diff
@@ -0,0 +1,88 @@
+package io.prometheus.client.exporter;
+
+/**
+ * Basic HTTP authenticator
+ */
+public class BasicAuthenticator extends com.sun.net.httpserver.BasicAuthenticator {
+
+    private static final String REALM = "/";
+
+    private String username;
+    private String password;
+
+    private BasicAuthenticator(String username, String password) {
+        super(REALM);
+        this.username = username;
+        this.password = password;
+    }
+
+    /**
+     * Check username and password
+     * @param username
+     * @param password
+     * @return
+     */
+    @Override
+    public boolean checkCredentials(String username, String password) {
+        if ((this.username == null) || (this.username.trim().length() == 0)) {
+            return false;
+        }
+
+        if ((this.password == null) || (this.password.trim().length() == 0)) {
+            return false;
+        }
+
+        return (this.username.equals(username) && this.password.equals(password));
+    }
+
+    /**
+     * BasicAuthenticator Builder
+     */
+    public static class Builder {
+
+        private String username;
+        private String password;
+
+        /**
+         * Required: username for the authenticator
+         * @param username
+         * @return
+         */
+        public Builder withUsername(String username) {
+            this.username = username;
+            return this;
+        }
+
+        /**
+         * Required: password for the authenticator
+         * @param password
+         * @return
+         */
+        public Builder withPassword(String password) {
+            this.password = password;
+            return this;
+        }
+
+        /**
+         * Build the authenticator
+         * @return
+         */
+        public BasicAuthenticator build() {;
+            assertNotNullNotEmpty(this.username, "username is null or empty");
+            assertNotNullNotEmpty(this.password, "password is null or empty");
```

Comment:
What about moving this check to the constructor and making the constructor public? Then users could choose whether they want to use the `Builder` or not.

Reply:
I will refactor to simplify the code.